waspc/src/Wasp/Generator/DbGenerator/Operations.hs

```diff
@@ -135,15 +149,32 @@ dbSeed genProjectDir seedName = do
     ExitSuccess -> Right ()
     ExitFailure c -> Left $ "Failed with exit code " <> show c
 
-isDbRunning ::
+testDbConnection ::
   Path' Abs (Dir ProjectRootDir) ->
-  IO Bool
-isDbRunning genProjectDir = do
+  IO DbConnectionTestResult
+testDbConnection genProjectDir = do
   chan <- newChan
   exitCode <- DbJobs.dbExecuteTest genProjectDir chan
-  -- NOTE: We only care if the command succeeds or fails, so we don't look at
-  -- the exit code or stdout/stderr for the process.
-  return $ exitCode == ExitSuccess
+
+  case exitCode of
+    ExitSuccess -> return DbConnectionSuccess
+    ExitFailure _ -> do
+      outputLines <- collectJobTextOutputUntilExitReceived chan
+      let databaseNotCreated = any containsDatabaseNotCreatedError outputLines
+
+      return $
+        if databaseNotCreated
+          then DbNotCreated
+          else DbConnectionFailure
+
+-- Prisma error code for "Database not created" is P1003.
+containsDatabaseNotCreatedError :: T.Text -> Bool
+containsDatabaseNotCreatedError text = text TR.=~ ("\\bP1003\\b" :: String)
+
+isDbConnectionEstablished :: DbConnectionTestResult -> Bool
```

Comment:
I get what you do with this function, makes sense! I wonder a bit about name -> so we call that "connection established"? Even if database does not exist? It is not a bad name, but I wonder if it is precise enough. I guess the problem is I am not sure what "db connection established" really means.
Is there a better matching name? Maybe `isDbInValidState`? or `isDbReady`? `isDbInUsableState`? `isDbReadyForPrisma`? Something in that direction? If we can't find anything better, fine, let's leave it as it is, but I do feel that `isDbConnectionEstablished` could maybe be better.
If nothing, you could go with `didDbConnectionTestSucceed`.

Reply:
I've went with `isDbConnectionPossible` since that's the check I was going for. The DB server can be connected to.